core/src/main/scala/kafka/network/BlockingChannel.scala

```diff
@@ -51,9 +51,9 @@ class BlockingChannel( val host: String,
     if(!connected) {
       try {
         channel = SocketChannel.open()
-        if(readBufferSize > 0)
+        if(readBufferSize != Selectable.USE_DEFAULT_BUFFER_SIZE)
           channel.socket.setReceiveBufferSize(readBufferSize)
-        if(writeBufferSize > 0)
+        if(writeBufferSize != Selectable.USE_DEFAULT_BUFFER_SIZE)
```

Comment:
This is a change in behaviour and `BlockingChannel` is deprecated, so I would prefer not to change it.